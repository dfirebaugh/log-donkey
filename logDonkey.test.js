const logDonkey = require("./logDonkey.js");

describe("Testing setLogLevel!", function(){
    it("should default to ERROR logLevel", function(){
        expect(logDonkey.getLogLevel()).toBe(logDonkey.levels.ERROR)
    })

    it("should be able to set logLevels to different levels", function(){
        Object.keys(logDonkey.levels).forEach(function checkSetLogLevel(level){
            logDonkey.setLogLevel(logDonkey.levels[level])
            expect(logDonkey.getLogLevel()).toBe(logDonkey.levels[level])
        })
    })
})
