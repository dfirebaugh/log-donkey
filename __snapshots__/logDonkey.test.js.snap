// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing basic logs should log debug 1`] = `
"debug: logging to debug,some more stuffz,

xxx
more debug here"
`;

exports[`Testing basic logs should log error 1`] = `"error: some stuff in here,    another thing"`;

exports[`Testing basic logs should log info 1`] = `
"info: logging to info,some more stuffz,


more info here"
`;
